server/server.go

```diff
@@ -19,63 +19,46 @@ var logger = monitor.NewModuleLogger("server")
 
 // Server holds entities that can be used to control the web server
 type Server struct {
-	defaultHeaders map[string]string
-	proxyService   *proxy.ProxyService
-	stopChan       chan os.Signal
-	stopWait       time.Duration
-	address        string
-	router         *mux.Router
-	listener       *http.Server
-}
-
-// Options holds basic web server settings.
-type Options struct {
-	Address         string
-	ProxyService    *proxy.ProxyService
-	StopWaitSeconds int
+	address  string
+	listener *http.Server
+	stopChan chan os.Signal
+	stopWait time.Duration
 }
 
 // NewServer returns a server initialized with settings from supplied options.
-func NewServer(opts Options) *Server {
-	s := &Server{
-		proxyService: opts.ProxyService,
-		address:      opts.Address,
-		stopWait:     15 * time.Second,
-		stopChan:     make(chan os.Signal),
-		defaultHeaders: map[string]string{
-			"Server":                      "api.lbry.tv",
-			"Access-Control-Allow-Origin": "*",
-		},
-	}
-	if opts.StopWaitSeconds != 0 {
-		s.stopWait = time.Duration(opts.StopWaitSeconds) * time.Second
-	}
-
+func NewServer(address string, sdkRouter *sdkrouter.Router) *Server {
 	r := mux.NewRouter()
-	api.InstallRoutes(s.proxyService, r)
+	api.InstallRoutes(r, sdkRouter)
 	r.Use(monitor.ErrorLoggingMiddleware)
-	r.Use(s.defaultHeadersMiddleware)
-	s.router = r
-
-	s.listener = &http.Server{
-		Addr:    s.address,
-		Handler: s.router,
-		// Can't have WriteTimeout set for streaming endpoints
-		WriteTimeout:      0,
-		IdleTimeout:       0,
-		ReadHeaderTimeout: 10 * time.Second,
+	r.Use(defaultHeadersMiddleware(map[string]string{
+		"Server":                      "api.lbry.tv",
+		"Access-Control-Allow-Origin": "*",
+	}))
+
+	return &Server{
+		address:  address,
+		stopWait: 15 * time.Second,
+		stopChan: make(chan os.Signal),
+		listener: &http.Server{
+			Addr:    address,
+			Handler: r,
+			// Can't have WriteTimeout set for streaming endpoints
```

Comment:
We don't have those anymore so setting at least a write timeout would be a good idea.